internal/contentmanager/contentmanager.go

```diff
@@ -0,0 +1,203 @@
+			// based on the GVK we need to use the unstructured types for mapping
+			u := &unstructured.Unstructured{}
+			u.SetGroupVersionKind(gvk)
+			ul := &unstructured.UnstructuredList{}
+			ul.SetGroupVersionKind(gvk.GroupVersion().WithKind(listKind))
+
+			scheme.AddKnownTypeWithName(gvk, u)
+			scheme.AddKnownTypeWithName(gvk.GroupVersion().WithKind(listKind), ul)
+			// Adding the common meta schemas to the scheme for the GroupVersion
+			// is necessary to ensure the scheme is aware of the different operations
+			// that can be performed against the resources in this GroupVersion
+			metav1.AddToGroupVersion(scheme, gvk.GroupVersion())
+		}
+	}
+
+	return scheme, nil
+}
+
+// ManageContent configures a controller-runtime cache.Cache and establishes watches for the provided resources.
+// It utilizes the provided ClusterExtension to set a DefaultLabelSelector on the cache.Cache
+// to ensure it is only caching and reacting to content that belongs to the ClusterExtension.
+// For each client.Object provided, a new source.Kind is created and used in a call to the Watch() method
+// of the provided controller.Controller to establish new watches for the managed resources.
+func (i *instance) ManageContent(ctx context.Context, ctrl controller.Controller, ce *v1alpha1.ClusterExtension, objs []client.Object) error {
+	// Return nil if the objs slice is empty before any further action
+	if len(objs) == 0 || ce == nil || ctrl == nil {
+		return nil
+	}
+
+	cfg, err := i.rcm(ctx, ce, i.baseCfg)
+	if err != nil {
+		return fmt.Errorf("getting rest.Config for ClusterExtension %q: %w", ce.Name, err)
+	}
+
+	scheme, err := buildScheme(objs)
+	if err != nil {
+		return fmt.Errorf(
+			"building scheme for %s; %w",
+			ce.GetName(),
+			err,
+		)
+	}
+
+	tgtLabels := labels.Set{
+		oclabels.OwnerKindKey: v1alpha1.ClusterExtensionKind,
+		oclabels.OwnerNameKey: ce.GetName(),
+	}
+
+	c, err := cache.New(cfg, cache.Options{
+		Scheme:               scheme,
+		DefaultLabelSelector: tgtLabels.AsSelector(),
+	})
+	if err != nil {
+		return fmt.Errorf("creating cache for ClusterExtension %q: %w", ce.Name, err)
+	}
+
+	for _, obj := range objs {
+		err = ctrl.Watch(
+			source.Kind(
+				c,
+				obj,
+				handler.TypedEnqueueRequestForOwner[client.Object](
+					scheme,
+					i.mapper,
+					ce,
+				),
+			),
+		)
+		if err != nil {
+			return fmt.Errorf("creating watch for ClusterExtension %q managed resource %s: %w", ce.Name, obj.GetObjectKind().GroupVersionKind(), err)
+		}
+	}
+
+	// TODO: Instead of stopping the existing cache and replacing it every time
+	// we should stop the informers that are no longer required
+	// and create any new ones as necessary. To keep the initial pass
+	// simple, we are going to keep this as is and optimize in a follow-up.
+	// Doing this in a follow-up gives us the opportunity to verify that this functions
+	// as expected when wired up in the ClusterExtension reconciler before going too deep
+	// in optimizations.
```

Comment:
This will be done in a follow-up

Reply:
If we hooked this up as is, what would end up triggering a cache teardown and re-creation? Would it be just when we call `actionclient.Upgrade`? Or would we do it even after `actionclient.Reconcile`?

I think if it is just for an upgrade, we can probably get by because that would only happen when the resolved bundle changes or (eventually when we support it) the user-supplied configuration changes.

Reply:
Hooked up as is, I would assume we only call this whenever the identified state is one of `Install` or `Upgrade`

Reply:
To clarify, by identified state I mean the same state that is used in a few other places in the `reconcile` function of the ClusterExtensionReconciler:
https://github.com/operator-framework/operator-controller/blob/db5a75c0e1e7b78ef98f2a29c2bc5550a509a6d1/internal/controllers/clusterextension_controller.go#L341-L354